src/msg/producer/writer/writer.go

```diff
@@ -174,9 +179,47 @@ func (w *writer) process(update interface{}) error {
 	for _, cs := range t.ConsumerServices() {
 		key := cs.ServiceID().String()
 		csw, ok := w.consumerServiceWriters[key]
+
 		if ok {
+			// update existing consumer service writer
+
 			csw.SetMessageTTLNanos(cs.MessageTTLNanos())
+
+			if cs.DynamicFilterConfigs() != nil {
+				dynamicFilters, err := ParseDynamicFilters(csw, cs.DynamicFilterConfigs())
+
+				if err != nil {
+					w.logger.Error("could not update dynamic filters on consumer service writer, error registering dynamic filters",
+						zap.String("writer", cs.String()), zap.Error(err))
+
+					multiErr = multiErr.Add(err)
+				} else {
+					// unregister all filters and register the new ones
+					w.Lock()
+
+					csw.UnregisterFilters()
+					for _, dynamicFilter := range dynamicFilters {
+						csw.RegisterFilter(dynamicFilter)
+					}
+
+					w.Unlock()
+				}
+			} else {
+				// sending no dynamic filters means we should remove all filters, unless there are static filters
+				// if there are no static filters, remove all filters
```

Comment:
actually probably makes sense to instead unregister filters, then apply static filters if there are any.